ui/projects/spline-api/src/lib/execution-event/services/execution-event.facade.ts

```diff
@@ -36,13 +36,13 @@ export class ExecutionEventFacade extends BaseFacade {
         super(http)
     }
 
-    fetchLineageOverview(executionEventId: string, maxDepth: number = 3): Observable<ExecutionEventLineageOverview> {
+    fetchLineageOverview<T = string>(executionEventId: string, overviewType: T, maxDepth: number = 3): Observable<ExecutionEventLineageOverview> {
```

Comment:
@wajda I've investigated the app architecture, and I concluded that spline-api shouldn't have any dependencies from the spline-common, due to the main purpose of this sub-project - connection to BE.
I've also considered to placed an enum in the spline-api and then using it in the spline-common, but it will be an incorrect way of the entity usage because this enum is a view representation, but not the API.
So, due to the reason I'm not sure would spline-api is used anywhere else, I just wanted to place some type for default usage of this certain function, so that is why you see 'string' there. So it means I could remove the default type if it bothers you.

Reply:
No, `string` is the least problem in my list:
1. Generic type `T` is completely redundant for both the method itself and the callers.
2. The effective type of the `overviewType` parameter from the perspective of the method body is always `any` (the default type value `string` has no effect - you can replace it with any possible type and it will compile)
3. As a consequence of the previous point, there is no control or suggestions of the values that can be passed to the `overviewType` parameter, while the implementation de facto expects only 2 possible values to yield correct result. That implies enum.

So we have to have some sort of enum here. And I'm not talking about that particular enum here. The point is that Spline Consumer REST API provides two resources that you can fetch, hence `ExecutionEventFacade` has to reflect it in one way or another. Whether it be a proper enum, or a boolean, or two different methods (e.g. `fetchLineageOverview` and `fetchImpactOverview`), it's a matter of choice.